torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1652,6 +1676,69 @@ def rekey_optim_state_dict(
             return new_osd
         return new_osd  # should never reach here
 
+    @staticmethod
+    def _optim_state_dict(
+        model: torch.nn.Module,
+        optim: torch.optim.Optimizer,
+        group: Optional[dist.ProcessGroup] = None,
+    ) -> Dict[str, Any]:
+        """
+        This API is still being developed, hence the `_` prefix. The comment
+        below is also not fully implemented yet. Do not use this API unless
+        you know why this API exists and how this API works.
+
+        Returns the optimizer state. The state will be sharded or consolidated
+        based on ``state_dict_type`` set by :meth:`set_state_dict_type` or
+        :meth:`state_dict_type`.
+
+        Args:
+            model (torch.nn.Module): Root module (which may or may not be a
+                :class:`FullyShardedDataParallel` instance) whose parameters
+                were passed into the optimizer ``optim``.
+            optim (torch.optim.Optimizer): Optimizer for ``model`` 's
+                parameters.
+                the default process group. (Default: ``None``)
+
+        Returns:
+            Dict[str, Any]: A :class:`dict` containing the optimizer state for
+            ``model``. The sharding of the optimizer state is based on
+            ``state_dict_type``.
+        """
+        return FullyShardedDataParallel.full_optim_state_dict(
+            model, optim, rank0_only=False, group=group
+        )
+
+    @staticmethod
+    def _load_optim_state_dict(
+        state_dict: Dict[str, Any],
+        model: torch.nn.Module,
+        optim: torch.optim.Optimizer,
+        group: Optional[dist.ProcessGroup] = None,
+    ) -> Dict[str, Any]:
+        """
+        This API is still being developed, hence the `_` prefix. The comment
+        below is also not fully implemented yet. Do not use this API unless
+        you know why this API exists and how this API works.
+
+        Load the optimizer state, ``state_dict``, to the optimizer ``optim``.
+        ``state_dict_type`` set by :meth:``set_state_dict_type`` decides how
+        to load the state_dict.
+
+        Args:
+            state_dict (Dict[str, Any]): The optimizer states to be loaded.
+            model (torch.nn.Module): Root module (which may or may not be a
+                :class:`FullyShardedDataParallel` instance) whose parameters
+                were passed into the optimizer ``optim``.
+            optim (torch.optim.Optimizer): Optimizer for ``model`` 's
+                parameters.
+            group (dist.ProcessGroup): Model's process group or ``None`` if using
```

Comment:
nit: For both here and `_optim_state_dict`, let us mention that this is the process group across which parameters are sharded since we have landed HSDP, which has both the intra-node sharding process group and the inter-node replicating process group.
```suggestion
            group (dist.ProcessGroup): Model's process group across which parameters are sharded or ``None`` if using
```